hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -58,132 +70,198 @@ public HoodieCommitArchiveLog(HoodieWriteConfig config, FileSystem fs) {
             .getArchiveLogPath(config.getBasePath() + "/" + HoodieTableMetaClient.METAFOLDER_NAME);
     }
 
-    /**
-     * Check if commits need to be archived. If yes, archive commits.
-     */
-    public boolean archiveIfRequired() {
-        List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
-        boolean success = true;
-        if (commitsToArchive.iterator().hasNext()) {
-            log.info("Archiving commits " + commitsToArchive);
-            archive(commitsToArchive);
-            success = deleteInstants(commitsToArchive);
-        } else {
-            log.info("No Commits to archive");
+    private HoodieLogFormat.Writer openWriter() {
+        try {
+            if(this.writer == null) {
+                return HoodieLogFormat.newWriterBuilder()
+                        .onParentPath(archiveFilePath.getParent())
+                        .withFileId(archiveFilePath.getName())
+                        .withFileExtension(HoodieArchivedLogFile.ARCHIVE_EXTENSION)
+                        .withFs(fs)
+                        .overBaseCommit("").build();
```

Comment:
this is very hacky to pass in a empty string here... hmmm. anyways.. we need a clean up pass over this later for sure.